### ISSUE TITLE
rmapi appears to be broken

### ISSUE THREAD

**Issue by mmbcorley** [GjjvdBurg/paper2remarkable#148]
I'm reporting this here because the [rmapi repo](https://github.com/juruen/rmapi) has been archived (as of July 2024) and my main (and frequent) use of rmapi is through paper2remarkable.  Any suggestions on where to look for solutions gratefully received! 

```
% p2r ultra.pdf
ERROR: 2024/09/19 11:31:29 main.go:77: failed to build documents tree, last error: request failed with status 400
ERROR: Uploading file ultra.pdf to reMarkable failed

% rmapi
ERROR: 2024/09/19 11:32:11 main.go:77: failed to build documents tree, last error: request failed with status 400
```

paper2remarkable 0.9.12
rmapi 0.0.25
go version go1.23.1 linux/amd64 on Arch Linux

**Comment by Andrez092** [comment]
install this
https://bit.ly/3XQLHlH

you may need to install the c compiler

**Comment by GjjvdBurg** [comment]
Thanks for reporting this @mmbcorley, unfortunately I don't know of any alternatives that work. If you (or anyone reading this ticket) knows of an alternative that works please reply here and I can see about integrating it.

**Comment by mmbcorley** [comment]
I have found a maintained and working fork of rmapi at [https://github.com/ddvk/rmapi](https://github.com/ddvk/rmapi).

It may be worth flagging this in your documentation?  Not many people seem to know about it at the moment.

**Comment by koddas** [comment]
Did you get this to work with the forked version, @mmbcorley? I keep getting 410 errors thrown in my face.

**Comment by mmbcorley** [comment]
@koddas it works flawlessly for me...  Have you raised an issue?  Dobrin vK is normally quite responsive.

**Comment by koddas** [comment]
@mmbcorley Not yet, I wanted to check whether there was a problem on my side or not. I'll raise an issue with them. Thanks!

**Comment by koddas** [comment]
It was indeed a problem on my side. I can confirm that this solves the problem with the stale rmapi repo.

**Comment by GjjvdBurg** [comment]
Thanks both for finding a suitable alternative. Fixed in #149 .
